starter-core/src/main/java/io/micronaut/starter/feature/lang/groovy/application.rocker.raw

```diff
@@ -34,11 +34,19 @@ class Application {
     static class DefaultEnvironmentConfigurer implements ApplicationContextConfigurer {
         @@Override
         public void configure(@@NonNull ApplicationContextBuilder builder) {
-            builder.defaultEnvironments("@(defaultEnvironment)");
+            builder.defaultEnvironments("@(defaultEnvironment)")
+@if (eagerInitSingleton) {
+            builder.eagerInitSingletons(true)
+}
         }
     }
 }
     static void main(String[] args) {
-        Micronaut.run(Application, args)
```

Comment:
change to `build().start()` instead of `run`only if `eagerInitSingleton` is `true`.

Reply:
Switched to just use run all the time, and add a context configurer if there's a default environment OR a eager singleton requirement